src/tests.rs

```diff
@@ -0,0 +1,147 @@
+#[cfg(test)]
+mod tests {
```

Comment:
Move this `mod tests` directly into `emitter.rs`. This will allow you to remove `pub(crate)` from `EventContainer`.